crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2227,6 +2235,11 @@ impl<'db> TypeInferenceBuilder<'db> {
         let right_ty = self.infer_expression(right);
 
         match (left_ty, right_ty, op) {
+            // When interacting with Todo, Any and Unknown should propagate (as if we fix this
+            // `Todo` in the future, the result would then become Any or Unknown, respectively.)
+            (Type::Any, Type::Todo, _) | (Type::Todo, Type::Any, _) => Type::Any,
+            (Type::Unknown, Type::Todo, _) | (Type::Todo, Type::Unknown, _) => Type::Unknown,
+
```

Comment:
the comment here is useful, but I think these cases here are fully covered by the branch immediately following this; I don't think you need to add any code to this method

```suggestion
            // When interacting with Todo, Any and Unknown should propagate (as if we fix this
            // `Todo` in the future, the result would then become Any or Unknown, respectively.)
```